modules/shared/src/rotary-loop.ts

```diff
@@ -0,0 +1,60 @@
+import debug from 'debug';
+
+export class RotaryLoop {
+  private runningPromise?: Promise<unknown>;
+  private stopResolve?: (value: unknown) => void;
+  private stopping = false;
+
+  private isRunning = () => this.runningPromise !== undefined;
```

Comment:
nit: prefer a getter here:

```suggestion
  private get isRunning(): boolean {
    return this.runningPromise !== undefined;
  }
```

Reply:
Changed in c25f957